src/Bridge/Doctrine/Orm/Util/QueryBuilderHelper.php

```diff
@@ -30,15 +30,15 @@ private function __construct()
     /**
      * Adds a join to the queryBuilder if none exists.
      */
-    public static function addJoinOnce(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $alias, string $association, string $joinType = null, string $conditionType = null, string $condition = null, string $originAlias = null): string
+    public static function addJoinOnce(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $alias, string $association, string $joinType = null, string $conditionType = null, string $condition = null, string $originAlias = null, string $newAlias = null): string
     {
         $join = self::getExistingJoin($queryBuilder, $alias, $association, $originAlias);
 
         if (null !== $join) {
             return $join->getAlias();
         }
 
-        $associationAlias = $queryNameGenerator->generateJoinAlias($association);
+        $associationAlias = $newAlias ?: $queryNameGenerator->generateJoinAlias($association);
```

Comment:
`$newAlias ?? $queryNameGenerator->generateJoinAlias($association)` ?

Reply:
> `$newAlias ?? $queryNameGenerator->generateJoinAlias($association)` ?

Yes, you are right. It's more accurate.